sklearn/preprocessing/_encoders.py

```diff
@@ -647,7 +661,31 @@ def _compute_drop_idx(self):
                     )
                 )
                 raise ValueError(msg)
-            return np.array(drop_indices, dtype=object)
+            drop_idx_internal = np.array(drop_indices, dtype=object)
+
+        # `_drop_idx_internal` are the categories to drop *after* the infrequent
+        # categories are grouped together. If there are no infrequent categories,
+        # then `drop_idx_` = `_drop_idx_internal` or drop is not set.
+        self._drop_idx_internal = drop_idx_internal
+
+        if not self._infrequent_enabled or drop_idx_internal is None:
+            self.drop_idx_ = self._drop_idx_internal
+        else:
+            # If there are infrequent categories, then we need to map `drop_idx`
+            # back to the categories seen in `self.categories_`
+            drop_idx_ = []
+            for feature_idx, drop_idx in enumerate(drop_idx_internal):
+                default_to_infrequent = self._default_to_infrequent_mappings[
+                    feature_idx
+                ]
+                if drop_idx is None or default_to_infrequent is None:
+                    orig_drop_idx = drop_idx
+                else:
+                    orig_drop_idx = np.where(default_to_infrequent == drop_idx)[0][0]
```

Comment:
Looking at the pattern, I think this is where I like to use `np.flatnonzero` to avoid those zero-indexing.

Reply:
Reading the documentation, it seems that NumPy advocates for using `nonzero` instead of `where` when we only pass the condition. So I assume that it makes sense to swtich to `np.flatnonzero`.